src/service/resources-impl.js

```diff
@@ -1454,15 +1454,25 @@ export class Resources {
           this.exec_.getSize() == 0 &&
           this.queue_.getSize() == 0 &&
           now > this.exec_.getLastDequeueTime() + 5000) {
-      let idleScheduledCount = 0;
+      let idleScheduledPreloadCount = 0;
+      let idleScheduledLayoutCount = 0;
       for (let i = 0; i < this.resources_.length; i++) {
         const r = this.resources_[i];
         if (r.getState() == ResourceState.READY_FOR_LAYOUT &&
                 !r.hasOwner() && r.isDisplayed()) {
-          dev().fine(TAG_, 'idle layout:', r.debugid);
-          this.scheduleLayoutOrPreload_(r, /* layout */ false);
-          idleScheduledCount++;
-          if (idleScheduledCount >= 4) {
+          const idleLayout = r.renderOnIdleOutsideViewport();
+          this.scheduleLayoutOrPreload_(
+              r, /* layout */ idleLayout);
+          if (idleLayout) {
+            if (idleScheduledLayoutCount++ < 4) {
+              this.scheduleLayoutOrPreload_(r, /* layout */true);
+            }
+          } else  {
+            if (idleScheduledPreloadCount++ < 4) {
+              this.scheduleLayoutOrPreload_(r, /* layout */false);
+            }
+          }
+          if (idleScheduledPreloadCount >= 4 && idleScheduledLayoutCount >= 4) {
```

Comment:
Probably an `addition >= 4`.

Reply:
Not sure I follow... do you mean I should rename idleScheduledLayoutCount to addition or that the summation of the two shouldn't be greater than 4?  I created separate variables as I didn't want the existence of preloads to prevent layouts.  Ideally layout would get preference but I don't want to iterate twice.  Thoughts?